File: leetcode-submissions/0953-reverse-only-letters/solution.java
class Solution {
    public String reverseOnlyLetters(String s) {
        //two pointers
        int left = 0;
        int right = s.length() - 1;
        char[] curr = s.toCharArray();

        while(left < right) {
            if(Character.isLetter(curr[left]) && Character.isLetter(curr[right])) {
                char temp = curr[left];
                curr[left] = curr[right];
                curr[right] = temp;
                left++;
                right--;
            } else if(Character.isLetter(curr[left])) {
                right--;
            } else {
                left++;
            }
        }
        return String.valueOf(curr);
    }
}

File: leetcode-submissions/0953-reverse-only-letters/solution.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        word_list = list(s)

        right = len(word_list) - 1
        left = 0

        while left < right:

            if ((ord(word_list[left]) not in range(65,91)) and (ord(word_list[left]) not in range(97,123))):
                left += 1
            elif ((ord(word_list[right]) not in range(65,91)) and (ord(word_list[right]) not in range(97,123))):
                right -= 1

            else:
                helper = word_list[left]
                word_list[left] = word_list[right]
                word_list[right] = helper

                left += 1
                right -= 1

        return "".join(word_list)
        
